src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -866,7 +869,8 @@ protected CloseableHttpResponse executeMethod(HttpHost host, HttpRequestBase htt
         httpMethod.setConfig(requestConfig.build());
 
         CloseableHttpResponse response = client.execute(host, httpMethod, requestContext);
-        while (response.getStatusLine().getStatusCode() == API_RATE_LIMIT_CODE) {
+        Random random = new Random();
+        for (int attempts = 0; response.getStatusLine().getStatusCode() == API_RATE_LIMIT_CODE; attempts++) {
```

Comment:
The minimum sleep should remain 5 seconds.  Once the limit is hit, the client needs to a wait at least on the order of seconds before trying again.  

The maximum around 2 or 3 minutes sounds good.   

We don't currently limit the number of attempts.  That shouldn't change.  We should hit the max and stay there.